### PR TITLE
Fixing tools and refs

### PR DESCRIPTION
**Pull request by cpholguera**
This PR removes the "#### Tools" sections at the bottom of both chapters 0x5b/6b
- done after checking that are referenced before.
- some links were moved to "References" as they are not tools anyway.
- all other missing "References" were also added to these chapters.
- if we would keep this list we would have to sync it constantly as the chapter changes and if we add more recommended tools in the "Recommended tools" section. 
- the "Recommended tools" section should serve as an overview of the recommended tools.
- other tools are not part of the "Recommended tools" section but they are mentioned inline, which is sufficient I would say. In case not, I would rather include them in the "Recommended tools" section.

In addition, this PR adds the missing Tools and references in 0x5c and 0x6c and fixes several typos, one dead link and several extra whitespaces in links.

### CODE REVIEW

**Comment by OMTGreleaser** [comment]
Results for commit 483740385fd70588c81dde8d87a9c33e868040d7: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05c-Reverse-Engineering-and-Tampering.md [hunk 1]
```diff
@@ -1742,12 +1742,26 @@ File-hiding is of course only the tip of the iceberg: you can accomplish a lot u
 
 ### References
 
-- Hacking Soft Tokens Paper by Bernhard Mueller - <https://packetstormsecurity.com/files/138504/HITB_Hacking_Soft_Tokens_v1.2.pdf>
-- OWASP MSTG Crackmes - <https://github.com/OWASP/owasp-mstg/tree/master/Crackmes>
+- Hacking Soft Tokens Paper - <https://packetstormsecurity.com/files/138504/HITB_Hacking_Soft_Tokens_v1.2.pdf>
```
Comment (sushi2k) [inline Document/0x05c-Reverse-Engineering-and-Tampering.md]: For papers please use the format as in the style guide 
https://github.com/OWASP/owasp-mstg/blob/master/style_guide.md

File: Document/0x06b-Basic-Security-Testing.md [hunk 2]
```diff
@@ -1484,26 +1484,12 @@ For information on disabling SSL Pinning both statically and dynamically, refer
...
-- iProxy - <https://iphonedevwiki.net/index.php/SSH_Over_USB>
-- Keychain-dumper - <https://github.com/ptoomey3/Keychain-Dumper/>
-- MobSF - <https://github.com/MobSF/Mobile-Security-Framework-MobSF>
-- Needle - <https://github.com/mwrlabs/needle>
-- Objection - <https://github.com/sensepost/objection>
-- Reverse Engineering tools for iOS Apps - <http://iphonedevwiki.net/index.php/Reverse_Engineering_Tools>
-- SSL Kill Switch 2 - <https://github.com/nabla-c0d3/ssl-kill-switch2>
-- Usbmuxd - <https://github.com/libimobiledevice/usbmuxd>
-- Wireshark - <https://www.wireshark.org/download.html>
-- Xcode - <https://developer.apple.com/xcode/>
```
Comment (sushi2k) [inline Document/0x06b-Basic-Security-Testing.md]: Why did you remove all of the tools? At least the one that are not RE related should stay here

File: Document/0x06c-Reverse-Engineering-and-Tampering.md [hunk 3]
```diff
@@ -618,3 +618,33 @@ Use the following approach to patch the JavaScript file:
...
+- ipainstaller - <https://cydia.saurik.com/package/com.slugrail.ipainstaller/>
+- Hopper Disassembler - <https://www.hopperapp.com/>
+- Optool - <https://github.com/alexzielenski/optool>
+- ios-deploy - <https://github.com/phonegap/ios-deploy>
+- Xcode command line developer tools - <https://railsapps.github.io/xcode-command-line-tools.html>
+- Frida - <https://www.frida.re>
+- Objection - <https://github.com/sensepost/objection>
+- Swizzler project - <https://github.com/vtky/Swizzler2/>
+- OWASP UnCrackable Apps for iOS - <https://github.com/OWASP/owasp-mstg/tree/master/Crackmes#ios>
+- Damn Vulnerable iOS App - <http://damnvulnerableiosapp.com/>
```
Comment (sushi2k) [inline Document/0x06c-Reverse-Engineering-and-Tampering.md]: Please order alphabetically

File: Document/0x05b-Basic-Security_Testing.md [hunk 4]
```diff
@@ -1748,37 +1749,18 @@ For information on disabling SSL Pinning both statically and dynamically, refer
...
-- OWASP ZAP - <https://www.owasp.org/index.php/OWASP_Zed_Attack_Proxy_Project>
-- QARK - <https://github.com/linkedin/qark/>
-- Radare2 - <https://rada.re/r/>
-- R2frida - <https://github.com/nowsecure/r2frida/>
-- SDK tools - <https://developer.android.com/studio/index.html#downloads>
-- SSLUnpinning - <https://github.com/ac-pm/SSLUnpinning_Xposed>
-- Wireshark - <https://www.wireshark.org/>
-- MobSF - <https://github.com/MobSF/Mobile-Security-Framework-MobSF>
-- Nathan - <https://github.com/mseclab/nathan>
-- Xposed - <https://www.xda-developers.com/xposed-framework-hub/>
```
[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x05b-Basic-Security_Testing.md]: Hi Sven, this is just a proposal. Please take a look at the beginning of the PR, I've explained them. If you think is better to keep these sections I'll just roll back those changes.

File: Document/0x06b-Basic-Security-Testing.md (hunk 2 above)
Comment (cpholguera) [reply Document/0x06b-Basic-Security-Testing.md]: Please see https://github.com/OWASP/owasp-mstg/pull/1328#discussion_r299238713

File: Document/0x05c-Reverse-Engineering-and-Tampering.md [hunk 5]
```diff
@@ -1742,12 +1742,26 @@ File-hiding is of course only the tip of the iceberg: you can accomplish a lot u
...
 - apkx - <https://github.com/b-mueller/apkx>
-- Frida - <https://www.frida.re/docs/android/>
+- Android NDK Downloads - <https://developer.android.com/ndk/downloads/index.html#stable-downloads>
+- smalidea plugin for IntelliJ - <https://github.com/JesusFreke/smali/wiki/smalidea>
+- APKTool - <https://ibotpeaches.github.io/Apktool/>
+- Radare2 - <https://www.radare.org>
+- Angr - <https://angr.io/>
+- JEB Decompiler - <https://www.pnfsoftware.com>
+- IDA Pro - <https://www.hex-rays.com/products/ida/>
+- VxStripper - <http://vxstripper.pagesperso-orange.fr>
```
[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x05c-Reverse-Engineering-and-Tampering.md]: done

File: Document/0x06c-Reverse-Engineering-and-Tampering.md (hunk 3 above)
Comment (cpholguera) [reply Document/0x06c-Reverse-Engineering-and-Tampering.md]: done

**Comment by OMTGreleaser** [comment]
Results for commit e7b9ce67c311684c85b6367f21bd13e9e304420c: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05c-Reverse-Engineering-and-Tampering.md (hunk 1 above)
Comment (cpholguera) [reply Document/0x05c-Reverse-Engineering-and-Tampering.md]: done, following: [shortname] [Author(s)], [Title] - Link

**Comment by OMTGreleaser** [comment]
Results for commit df57f21661625762630a49632bf057f325ea02c0: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05b-Basic-Security_Testing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sushi2k) [reply Document/0x05b-Basic-Security_Testing.md]: I think it's better if we keep the links to tools at the end of 5b and 6b. Background is that the tools are linked in the text, but in the printed book the URL is not visible and it cannot be clicked. That's why we added the tools at the end of the section so it's also available for readers of the book.

File: Document/0x05c-Reverse-Engineering-and-Tampering.md (hunk 1 above)
Comment (sushi2k) [reply Document/0x05c-Reverse-Engineering-and-Tampering.md]: 👍

File: Document/0x05b-Basic-Security_Testing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x05b-Basic-Security_Testing.md]: Aha! I see, ok then I'll restore, go the other way round and check if there are some missing ones. Thanks for the clarification ;)

File: Document/0x06b-Basic-Security-Testing.md (hunk 2 above)
Comment (cpholguera) [reply Document/0x06b-Basic-Security-Testing.md]: done + extended with all missing tools

**Comment by OMTGreleaser** [comment]
Results for commit 50d98f15777c8b783d5f78755b257d053ad7c36f: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05b-Basic-Security_Testing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x05b-Basic-Security_Testing.md]: done + extended with all missing tools

**Comment by OMTGreleaser** [comment]
Results for commit 46601115181001c11d58a6a3f90c624434a60651: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.
